testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/RuntimeJournalTypeManagementTestCase.java

```diff
@@ -0,0 +1,128 @@
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.manualmode.messaging;
+
+import static org.jboss.as.controller.client.helpers.ClientConstants.VALUE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+import org.jboss.arquillian.container.test.api.ContainerController;
+
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.dmr.ModelNode;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test reading the journal runtime type.
+ *
+ * @author Emmanuel Hugonnet (c) 2019 Red Hat, Inc.
+ */
+@RunAsClient()
+@RunWith(Arquillian.class)
+public class RuntimeJournalTypeManagementTestCase {
+
+    private static final String DEFAULT_FULL_JBOSSAS = "default-full-jbossas";
+
+    private static final ModelNode SERVER_ADDRESS = PathAddress.parseCLIStyleAddress("/subsystem=messaging-activemq/server=default").toModelNode();
+
+    @ArquillianResource
+    protected static ContainerController container;
+
+    @Test
+    public void testReadFileJournaltype() throws IOException {
+        if (!container.isStarted(DEFAULT_FULL_JBOSSAS)) {
+            container.start(DEFAULT_FULL_JBOSSAS);
+        }
+        ManagementClient managementClient = createManagementClient();
+        assertEquals("ASYNCIO", readJournalType(managementClient));
+        assertNotNull(readRuntimeJournalType(managementClient));
+        ModelNode op = Operations.createAddOperation(Operations.createAddress("system-property", "org.apache.activemq.artemis.core.io.aio.AIOSequentialFileFactory.DISABLED"));
+        op.get(VALUE).set(true);
+        execute(managementClient, op, true);
+        managementClient.close();
+        container.stop(DEFAULT_FULL_JBOSSAS);
+        container.start(DEFAULT_FULL_JBOSSAS);
+        managementClient = createManagementClient();
+        assertEquals("ASYNCIO", readJournalType(managementClient));
+        assertEquals("NIO", readRuntimeJournalType(managementClient));
+        container.stop(DEFAULT_FULL_JBOSSAS);
+    }
+
+    @Test
+    public void testReadDatabaseJournaltype() throws IOException {
+        if (!container.isStarted(DEFAULT_FULL_JBOSSAS)) {
+            container.start(DEFAULT_FULL_JBOSSAS);
+        }
+        ManagementClient managementClient = createManagementClient();
+        assertEquals("ASYNCIO", readJournalType(managementClient));
+        assertNotNull(readRuntimeJournalType(managementClient));
```

Comment:
Same as above

Reply:
same as above, we can't presume of the value